dod-advana/gamechanger-web
# Task/UOT-130367 - Graph View Node Limits

## Description


- 3 new backend environment variables `PULL_NODES_FROM_NEO4J_MAX_LIMIT` `GRAPH_VIEW_NODES_DISPLAYED_WARNING_LIMIT` `MAX_GRAPH_VIEW_NODES_DISPLAYED`
- By default every graph view search is capped at `GRAPH_VIEW_NODES_DISPLAYED_WARNING_LIMIT` with a warning banner at the top with a "Load All" button
- "Load All" button pulls in nodes capped at `MAX_GRAPH_VIEW_NODES_DISPLAYED`
- Nodes are returned sorted by page rank
- Better align context menu around nodes

Warning with "Load All" button:
![image](https://user-images.githubusercontent.com/85301886/152420116-5a09d98d-55c6-445b-be65-bdeb539d43ad.png)

Max limit reached:
![image](https://user-images.githubusercontent.com/85301886/152420086-669e00d0-b15e-4566-8a32-9dcc8ce0508f.png)

Limits in pictures are set low for testing purposes, should be set higher in dev/prod

## Related Issue/Ticket




https://jira.di2e.net/browse/UOT-130367
https://jira.di2e.net/browse/UOT-130369

## Testing instructions



1. Set the new env vars in backend .env. Good values would be 
```
PULL_NODES_FROM_NEO4J_MAX_LIMIT=500
GRAPH_VIEW_NODES_DISPLAYED_WARNING_LIMIT=500
MAX_GRAPH_VIEW_NODES_DISPLAYED=5000
```
2. Pull up a graph view search with results fewer than `GRAPH_VIEW_NODES_DISPLAYED_WARNING_LIMIT`. Warning should not appear.
3. Pull up a search with results between `GRAPH_VIEW_NODES_DISPLAYED_WARNING_LIMIT` and `MAX_GRAPH_VIEW_NODES_DISPLAYED`. Warning should appear.
4. Click "Load All" button. All nodes should load in, warning should disappear.
5. Pull up a search with more results than `MAX_GRAPH_VIEW_NODES_DISPLAYED` (e.g. Navy). Warning should appear.
6. Click "Load All" button. `MAX_GRAPH_VIEW_NODES_DISPLAYED` nodes should load in, new max cap alert should appear.
